src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs

```diff
@@ -68,5 +72,28 @@ public static void DirectoryCopy(string sourceDirPath, string destDirPath, bool
                 }
             }
         }
+
+        public static IEnumerable<string> GetProjectPathsFromSolutionFile(string solutionPath)
+        {
+            if (solutionPath.Contains(".sln") && File.Exists(solutionPath))
```

Comment:
Hmm this would only occur in the event the solution file doesn't exist at all or is not a real file which should be very rare, falling back on old method or searching for .csproj files in that case is also not ideal as even attempting to go back up to the parent directory from a non existing sln file is somewhat tricky. The more likely case which is that the parsing fails has a proper fall back method or returning to directory scanning. I do agree however that we should log some type of warning or error however will add that in.